core/src/main/java/jenkins/util/JSONSignatureValidator.java

```diff
@@ -73,7 +75,17 @@ public FormValidation verifySignature(JSONObject o) throws IOException {
                     certs.add(c);
                 }
 
-                CertificateUtil.validatePath(certs, loadTrustAnchors(cf));
+                try {
+                    CertificateUtil.validatePath(certs, loadTrustAnchors(cf));
+                } catch (GeneralSecurityException | IOException e) {
+                    Iterable<Object> certSubjectDNs = Iterables.transform(certs, new Function<X509Certificate, Object>() {
+                        @Override
+                        public Object apply(X509Certificate cert) {
+                            return cert == null ? null : cert.getSubjectDN();
+                        }
+                    });
+                    return FormValidation.warning(e,String.format("Certificate validation failed with certs %s in %s", certSubjectDNs.toString(), name));
```

Comment:
The context of the exception is invaluable (will contain the reason validation failed).
It must at least be logged - and I would be tempted to included it in the validation result so that the user sees it.  (this will also fix the loss of information I mentioned above.)